src/components/AgreementLinesFieldArray/AgreementLineField.test.js

```diff
@@ -0,0 +1,1115 @@
+        await KeyValue('Name').has({ value: 'EResourceLink' });
+        await KeyValue('Publication type').has({ value: 'Journal' });
+        await KeyValue('Titles').has({ value: 'EResourceCount' });
+        await KeyValue('Provider').has({ value: 'EResourceProvider' });
+        await KeyValue('Default coverage').exists();
+      });
+
+      test('renders the CoverageFieldArray component', () => {
+        const { getByText } = renderComponent;
+        expect(getByText('CoverageFieldArray')).toBeInTheDocument();
+      });
+
+      test('renders the POLinesFieldArray component', () => {
+        const { getByText } = renderComponent;
+        expect(getByText('POLinesFieldArray')).toBeInTheDocument();
+      });
+    });
+
+    describe('renders expected fields/values with detached resource', () => {
+      beforeEach(() => {
+        renderComponent = renderWithIntl(
+          <TestForm onSubmit={onSubmit}>
+            <AgreementLineField
+              index={0}
+              input={{ name:'resource' }}
+              meta={detachedData.meta}
+              resource={detachedData.resource}
+            />
+          </TestForm>,
+          translationsProperties
+        );
+      });
+
+      it('renders description keyValue', async () => {
+        await KeyValue('Description').has({ value: 'agreement line description' });
+      });
+
+      test('renders expected fields', () => {
+        const { getByTestId, getByRole } = renderComponent;
+          expect(getByTestId('agreementLineField')).toBeInTheDocument();
+          expect(getByRole('textbox', { name: /active from/i })).toBeInTheDocument();
+          expect(getByRole('textbox', { name: /active to/i })).toBeInTheDocument();
+          expect(getByRole('textbox', { name: /note/i })).toBeInTheDocument();
+      });
+    });
+
+    describe('renders expected fields/values with initial values set', () => {
+        test('renders expected fields', () => {
+            const { getByTestId, getByRole } = renderWithIntl(
+              <TestForm initialValues={{ data }} onSubmit={onSubmit}>
+                <AgreementLineField
+                  index={0}
+                  input={{ name:'resource' }}
+                  meta={data.meta}
+                  resource={data.resource}
+                />
+              </TestForm>,
+          translationsProperties
+        );
+
+        expect(getByTestId('agreementLineField')).toBeInTheDocument();
+        expect(getByRole('textbox', { name: /active from/i })).toBeInTheDocument();
+        expect(getByRole('textbox', { name: /active to/i })).toBeInTheDocument();
+        expect(getByRole('textbox', { name: /note/i })).toBeInTheDocument();
+      });
+
+      test('date validation fires for invalid end date', async () => {
+        const { getAllByText, queryByText, getByRole } = renderWithIntl(
+          <TestForm initialValues={{ data }} onSubmit={onSubmit}>
+            <AgreementLineField
```

Comment:
The component here should be rendered via a <Field /> so that we dont have to manually pass the `input` and `meta` props manually. Please look at other example for similar patterns.